src-test/src/com/etendoerp/etendorx/datasource/ManageEntityFieldsDSJavaMappingTest.java

```diff
@@ -0,0 +1,177 @@
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.etendoerp.etendorx.TestUtils;
+import com.etendoerp.etendorx.data.ETRXJavaMapping;
+import com.etendoerp.etendorx.data.ETRXProjectionEntity;
+
+/**
+ * Test class for the ManageEntityFieldsDS class, focusing on the getJavaMappingFilterData method.
+ * This class uses Mockito for mocking dependencies and testing the method's behavior.
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class ManageEntityFieldsDSJavaMappingTest {
+
+  @InjectMocks
+  private ManageEntityFieldsDS manageEntityFieldsDS;
+
+  @Mock
+  private OBDal obDal;
+
+  @Mock
+  private Session session;
+
+  @Mock
+  private Query<ETRXJavaMapping> query;
+
+  @Mock
+  private ETRXJavaMapping javaMapping1;
+
+  @Mock
+  private ETRXJavaMapping javaMapping2;
+
+  private Method getJavaMappingFilterDataMethod;
+
+  /**
+   * Sets up the test environment by using reflection to access the private
+   * getJavaMappingFilterData method of the ManageEntityFieldsDS class.
+   *
+   * This method is called before each test method to prepare the test fixture
+   * by making the private method accessible for testing.
+   *
+   * @throws Exception if method reflection fails, such as if the method cannot be found
+   */
+  @Before
+  public void setUp() throws Exception {
+    getJavaMappingFilterDataMethod = ManageEntityFieldsDS.class.getDeclaredMethod(
+        "getJavaMappingFilterData",
+        ETRXProjectionEntity.class
+    );
+    getJavaMappingFilterDataMethod.setAccessible(true);
+  }
+
+  /**
+   * Tests the getJavaMappingFilterData method when Java mappings are found.
+   * Verifies that:
+   * - The method returns a list of maps with correct data
+   * - Each map contains the expected keys and values
+   * - Admin mode is set and restored correctly
+   *
+   * @throws Exception if reflection or method invocation fails
+   */
+  @Test
+  public void testGetJavaMappingFilterDataWithResults() throws Exception {
+    ETRXProjectionEntity projectionEntity = mock(ETRXProjectionEntity.class);
+    when(projectionEntity.getId()).thenReturn("test-projection-id");
+
+    when(javaMapping1.getId()).thenReturn("java-mapping-1");
+    when(javaMapping1.getIdentifier()).thenReturn(TestUtils.JAVA_MAPPING_1);
+
+    when(javaMapping2.getId()).thenReturn("java-mapping-2");
+    when(javaMapping2.getIdentifier()).thenReturn(TestUtils.JAVA_MAPPING_2);
+
+    List<ETRXJavaMapping> mockJavaMappings = Arrays.asList(javaMapping1, javaMapping2);
+
+    try (MockedStatic<OBContext> obContextMock = mockStatic(OBContext.class)) {
+      try (MockedStatic<OBDal> obDalMock = mockStatic(OBDal.class)) {
+        obDalMock.when(OBDal::getInstance).thenReturn(obDal);
+        when(obDal.getSession()).thenReturn(session);
+        when(session.createQuery(anyString(), eq(ETRXJavaMapping.class))).thenReturn(query);
+        when(query.setParameter(eq("etrxProjectionEntityId"), any())).thenReturn(query);
```

Comment:
`etrxProjectionEntityId `
I recommend passing to utils as they are used several times